tools/nimgrep.nim

```diff
@@ -33,13 +33,21 @@ Options:
   --word, -w          the match should have word boundaries (buggy for pegs!)
   --ignoreCase, -i    be case insensitive
   --ignoreStyle, -y   be style insensitive
+  --nWorkers:N, -n:N  speed up search by N additional workers (threads)
   --ext:EX1|EX2|...   only search the files with the given extension(s),
                       empty one ("--ext") means files with missing extension
   --noExt:EX1|...     exclude files having given extension(s), use empty one to
                       skip files with no extension (like some binary files are)
-  --includeFile:PAT   include only files whose names match the given regex PAT
-  --excludeFile:PAT   skip files whose names match the given regex pattern PAT
-  --excludeDir:PAT    skip directories whose names match the given regex PAT
+  --includeFile:PAT   search only files whose names match the given PATttern
+  --excludeFile:PAT   skip files whose names match the given pattern PAT
+  --includeDir:PAT    search only files with full directory name matching PAT
+  --excludeDir:PAT    skip directories whose names match the given pattern PAT
+  --if,--ef,--id,--ed abbreviations of 4 options above
+  --match:PAT         select files containing a (not displayed) match of PAT
+  --noMatch:PAT       select files not containing any match of PAT
+  --bin:yes|no|only   process binary files? (detected by \0 in first 1K bytes)
```

Comment:
use `on|off|only` simply for consistency with nim command line options and standard nim tools